packages/ui-components/src/components/UITreeDropdown/UITreeDropdown.tsx

```diff
@@ -424,19 +424,21 @@ export class UITreeDropdown extends React.Component<UITreeDropdownProps, UITreeD
     handleOnChangeValue = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
         const query = event.target as HTMLInputElement;
 
-        const list = this.state.originalItems.filter((item) => this.filterElement(query.value, item));
+        this.setState((prevState) => {
+            const newState = {
+                hasSelected: false,
+                value: query.value,
+                items: prevState.originalItems.filter((item) => this.filterElement(query.value, item)),
+                query: query.value,
+                valueChanged: true
+            } as UITreeDropdownState;
 
-        this.setState({
-            hasSelected: false,
-            value: query.value,
-            items: list,
-            query: query.value,
-            valueChanged: true
+            if (!prevState.isMenuOpen) {
+                this.toggleMenu(false);
+                newState.isMenuOpen = false;
+            }
+            return newState;
         });
-
-        if (!this.state.isMenuOpen) {
-            this.toggleMenu(false);
-        }
     };
```

Comment:
I see following error:
![image](https://github.com/SAP/open-ux-tools/assets/90789422/7b3d8065-5722-47c0-be47-5d98edff08a8)

Scenario:
1. Open storybook for tree drpdown;
2. Use tab key to navigate tree dropdown input - menu should not be opened;
3. Press char like `a` - it should trigger menu open;

I tried apply old version - I did not see such error

Reply:
@slin-sap
adjusted code with following - 0f92ca10ad71507f591599460c3cb919c755fb99
please correct if it provides issues